src/library/solrize.py

```diff
@@ -126,69 +168,38 @@ def process_hash_list(document_datasets):
 
             db.updateSolrizeStartDate(conn, file_id)
 
-            logger.info('Removing all docs for doc with hash: ' +
-                        file_hash + ' and id: ' + file_id)
-
-            for core_name in solr_cores:
-                try:
-                    solr_cores[core_name].delete(
-                        q='iati_activities_document_id:' + file_id)
-                except Exception as e:
-                    e_message = e.args[0] if hasattr(e, 'args') else ''
-                    raise SolrError('DELETING hash: ' + file_hash + ' and id: ' + file_id +
-                                    ', from collection with name ' + core_name + ': ' + e_message)
-
-            logger.info('Adding docs for hash: ' + file_hash + ' and id: ' + file_id)
+            # check whether existing DS data and new dataset contain only "good" data:
+            # - flattened activities have any duplicated IATI identifiers in them?
+            # - existing docs in solr have repeated IDs
+            new_identifiers = [fa['iati_identifier'] for fa in flattened_activities[0]]
+            has_dupes = len(set(new_identifiers)) != len(new_identifiers)
+            has_dupes = has_dupes or solr_cores['activity'].search("id:*--1", rows=0).hits > 0
```

Comment:
I don't understand how this works - doesn't it need to search limited by the file-id too?

Reply:
Yeah, whoops. I had that check in and lost it somewhere along the lines in a refactor.